src/microbe_stage/BioProcess.cs

```diff
@@ -64,6 +64,25 @@ public void Check(string name)
         TranslationHelper.CopyTranslateTemplatesToTranslateSource(this);
     }
 
+    public float GetRateWithConditions(BiomeConditions conditions)
+    {
+        var rate = 1.0f;
+
+        foreach (var input in Inputs)
+        {
+            if (input.Key.IsEnvironmental)
+            {
+                foreach (var compound in conditions.AverageCompounds)
+                {
+                    if (compound.Key.Equals(input.Key) && rate > 0)
+                        rate = compound.Value.Ambient / rate;
```

Comment:
I'm also not entirely sure about this formula. If its purpose is to scale the process rate with compound concentrations, then there should probably be multiplication instead of division.
For example, a compound concentration of 0.01 would increase the rate 100x on the next iteration, while it is supposed to actually decrease it 100x, if I understand it correctly.

Reply:
I'm not from the first rows when it comes to making good algorithms, what you or someone else could propose?

Reply:
Something like this:
````
if (compound.Key.Equals(input.Key))
     rate *= compound.Value.Ambient;
````

Reply:
Oh right